src/Relude/Monad/Either.hs

```diff
@@ -112,46 +112,86 @@ maybeToLeft :: r -> Maybe l -> Either l r
 maybeToLeft r = maybe (Right r) Left
 {-# INLINE maybeToLeft #-}
 
--- | Applies given action to 'Either' content if 'Left' is given and returns
+{- | Applies given action to 'Either' content if 'Left' is given and returns
 -- the result. In case of 'Right' the default value will be returned.
+
+>>> whenLeft "bar" (Left "foo") (\a -> [a])
+["foo"]
+
+>>> whenLeft "bar" (Right "foo") (\a -> [a])
+["bar"]
+
+-}
 whenLeft :: Applicative f => a -> Either l r -> (l -> f a) -> f a
 whenLeft _ (Left  l) f = f l
 whenLeft a (Right _) _ = pure a
 {-# INLINE whenLeft #-}
 
--- | Applies given action to 'Either' content if 'Left' is given.
+{- | Applies given action to 'Either' content if 'Left' is given.
+
+TODO: whenLeft_
+
+-}
 whenLeft_ :: Applicative f => Either l r -> (l -> f ()) -> f ()
 whenLeft_ = whenLeft ()
 {-# INLINE whenLeft_ #-}
 
--- | Monadic version of 'whenLeft'.
+{- | Monadic version of 'whenLeft'.
+
+TODO: whenLeftM
```

Comment:
Hard to come up with real-life example of value with type `m (Either l r)`. So, for now, you can use the same examples as in any other code, just wrap `Either` in `pure`